### PR TITLE
Description feature

### PR DESCRIPTION
**Pull request by fabe85**
@jonas-he 
@swathi-ssunder 
@samitsv 
@vivek-sethia

### CODE REVIEW

File: routes.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ module.exports = function (app, router) {
     //route for getting pokemon details by specific name
     router.get('/pokemon/name/:name', pokemon.getByName);
 
+    //route for getting pokemon details by description
+    router.get('/pokemon/description/:description', pokemon.getByDescription());
```
Comment (AlexanderLill) [inline routes.js]: I think this needs to be without "()" and implemented in
https://github.com/PokemonGoers/PokeData/blob/8208525bbac1c592e1d6d1a651ee8e51daf19c05/app/stores/pokemon.js ?

Can this be implemented like the "getByName" so we can e.g. search for "Seed" and get "Bulbasaur" as result with "can be seen napping in bright sunlight. There is a seed..."? :)

File: app/api/pokemon.js [hunk 2]
```diff
@@ -161,6 +161,28 @@ module.exports = {
...
+     * @apiGroup Pokemon
+     * @apiParam {String} description Description of the pokemon
+     *
+     * @apiUse SamplePokemon
+     * @apiUse NoRecords
+     *
+     */
+    getByDescription: function(req, res) {
+        logger.info('Get Pokemon description');
+        pokemon.getByDescription(req.params.name, function (success, message) {
```
Comment (AlexanderLill) [inline app/api/pokemon.js]: "name" should be replaced by "description" if I understand that right, also in the error message

Comment (fabe85) [reply app/api/pokemon.js]: Done 👍

File: routes.js (hunk 1 above)
Comment (fabe85) [reply routes.js]: Done 👍 
Please let me know if the search has to be adjusted

Comment (AlexanderLill) [reply routes.js]: Will do, thanks :)

File: app/stores/pokemon.js [hunk 3]
```diff
@@ -129,6 +129,26 @@ module.exports = {
...
+                    } else {
+                        callback(status, response);
+                    }
+                }.bind(this));
+            } else {
+                callback(status, response);
+            }
+        }.bind(this));
+    },
+    /*
```
Comment (swathi-ssunder) [inline app/stores/pokemon.js:177]: Is it required to do a partial search based on hamming algorithm for description too?  Maybe just the regex based search is sufficient.

Comment (swathi-ssunder) [reply app/stores/pokemon.js:177]: Also, if partial search needs to be done, changes need to be made. Currently, in getClosest, **name** attribute is being used.
So in the current state, this api for partial search by description wont work anyways.

Comment (fabe85) [reply app/stores/pokemon.js:177]: You are right. If you look at the comment above it is indeed preferable to have this feature.

File: routes.js (hunk 1 above)
Comment (fabe85) [reply routes.js]: @AlexanderLill Please let me know if the search feature is ok for your purpose

Comment (AlexanderLill) [reply routes.js]: Unfortunately i cannot test the API any more as it seems the develop was pushed to docker now (and not this branch with the description feature) but i think it should be fine like this!

**Pull request merged**